### PR TITLE
Allow tests to vary by Grisu/Ryu

### PR DESCRIPTION
**Pull request by timholy**
CC @isentropic. Does this seem OK?

### CODE REVIEW

File: src/Showoff.jl [hunk 1]
```diff
@@ -74,7 +74,7 @@ function concrete_maximum(xs)
 end
 
 function scientific_precision_heuristic(xs::AbstractArray{<:AbstractFloat})
-    ys = [x == 0.0 ? 0.0 : x / 10.0^floor(log10(abs(x)))
+    ys = [x == 0.0 ? 0.0 : round(10.0 ^ (z = log10(abs(Float64(x))); z - floor(z)); sigdigits=15)
```
Comment (timholy) [inline src/Showoff.jl:77]: On my CPU the original gave 1 ulp of error, which Ryu cares about but which Grisu ignored. But that's probably CPU-dependent.

Comment (isentropic) [reply src/Showoff.jl:77]: Perhaps there is a cleaner way to do it?
https://stackoverflow.com/questions/15685181/how-to-get-the-sign-mantissa-and-exponent-of-a-floating-point-number
Like somehow extract the underlying exponent of a float. I'm not really into low-level basics of Julia, so perhaps you know if there is an easy way.

If not, I think this is fine

**Review by isentropic — APPROVED** [review]
Looks good

File: src/Showoff.jl (hunk 1 above)
Comment (timholy) [reply src/Showoff.jl:77]: For base 2 there is, but this is base 10. I don't think there is.

**Pull request merged**